src/fillTank.test.js

```diff
@@ -1,11 +1,100 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared ', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('should calculate full tank If the amount is not given', () => {
+    const customer = {
+      money: 3000, // customer account balance
+      vehicle: {
+        maxTankCapacity: 40, // fuel tank volume
+        fuelRemains: 0, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 30);
 
+    expect(customer.money).toBe(1800);
+    expect(customer.vehicle.fuelRemains).toBe(40);
   });
 
-  // write tests here
+  it(
+    'should calculate full tank If the amount is greater than the tank can use',
+    () => {
+      const customer = {
+        money: 3000, // customer account balance
+        vehicle: {
+          maxTankCapacity: 20, // fuel tank volume
+          fuelRemains: 5, // Remaining fuel in the tank
+        },
+      };
+
+      fillTank(customer, 50, 20);
+
+      expect(customer.money).toBe(2250);
+      expect(customer.vehicle.fuelRemains).toBe(20);
+    });
+
+  it('should fill the fuel if the client can pay for it', () => {
+    const customer = {
+      money: 200, // customer account balance
+      vehicle: {
+        maxTankCapacity: 30, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 20, 20);
+
+    expect(customer.money).toBe(0);
+    expect(customer.vehicle.fuelRemains).toBe(15);
+  });
+
+  it('should round the amount by discarding number to the tenth part.', () => {
+    const customer = {
+      money: 600, // customer account balance
+      vehicle: {
+        maxTankCapacity: 50, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 9, 10.239);
+
+    expect(customer.vehicle.fuelRemains).toBe(15.2);
+  });
+
+  it('should not pour at all if the amount is less than 2 liters', () => {
+    const customer = {
+      money: 375, // customer account balance
+      vehicle: {
+        maxTankCapacity: 30, // fuel tank volume
+        fuelRemains: 5, // Remaining fuel in the tank
+      },
+    };
+
+    fillTank(customer, 25, 1.5);
+
+    expect(customer.money).toBe(375);
+    expect(customer.vehicle.fuelRemains).toBe(5);
```

Comment:
The test case expects no fuel to be poured if the amount is less than 2 liters, which is correct. Ensure the `fillTank` function correctly implements this logic.